### DISCUSSION TITLE
Setting the seed on resample()

### DISCUSSION THREAD

**Discussion by yanivabir** [mwshinn/PyDDM#47] [Ask for help] [answered]
Hi,

I'm a little confused by what the seed argument of the resample function does.

I expected that if I set the seed, I will get identical simulated data every time I run resample on a model. But that does not seem to be the case?

What does seed do, and is there a way to fix the random seed such that I get exactly the same data every time I run resample?

Many thanks!
Yaniv

**Comment by mwshinn** [comment, marked as answer]
This was a bug and I just fixed it with commit ffc1c793437959bd80e7a59c7e7412883a4970e2.  Thanks for pointing this out!  Since the latest version of PyDDM has some big changes, if you want a workaround without installing the latest version, run:

```
prev_state = np.random.get_state()
np.random.seed(1)
(Call to resample)
np.random.set_state(prev_state)
```

**Reply by yanivabir** [reply]
thanks!
